pkg/cluster/cluster.go

```diff
@@ -72,7 +75,8 @@ type manager struct {
 	virtualMachines       compute.VirtualMachinesClient
 	interfaces            network.InterfacesClient // TODO: use armInterfaces instead.
 	armInterfaces         armnetwork.InterfacesClient
-	publicIPAddresses     network.PublicIPAddressesClient
+	publicIPAddresses     network.PublicIPAddressesClient // TODO: use armPublicIPAddresses instead.
+	armPublicIPAddresses  armnetwork.PublicIPAddressesClient
```

Comment:
That is because it would be a bigger PR if we remove it in this PR.
For example, `publicIPAddresses` is used in `createOrUpdateRouterIPEarly`, `deleteUnusedManagedIPs`, `getClusterManagedIPs`, and `createPublicIPAddress`.
To make the PR simple and make the change scope as small as possible, I keep the old one.

I added the statement about that to the description. thanks!

Reply:
Perfect! That is what I thought, just wanted to be sure about it. Thank you a lot. 🙂